### PR TITLE
 Add control plane first stone: a gRPC metastore service.

### PR DESCRIPTION
**Pull request by fmassot**
### Description

Add a metastore service with 2 implementations:
- One backed by a `Metastore` instance.
- One backed by a gRPC client.

### Important note

The PR still needs a lot of cleaning and testing but I think the API is good so ready to have some feedback about the API.

### CODE REVIEW

File: quickwit-control-plane/src/index.rs [hunk 1]
```diff
@@ -0,0 +1,240 @@
+// Copyright (C) 2022 Quickwit, Inc.
```
Comment (fmassot) [inline quickwit-control-plane/src/index.rs:1]: This is essentially a copy/paste from the functions present in `quickwit-core`

File: Cargo.toml [hunk 2]
```diff
@@ -7,9 +7,9 @@ members = [
   "quickwit-cluster",
   "quickwit-common",
   "quickwit-config",
-  "quickwit-core",
```
Comment (fmassot) [inline Cargo.toml:10]: `quickwit-core` did not has really functionnal meaning so removing it is a good thing as now we have a place where we can put all stuff relative to index operations (control plane).

File: quickwit-cli/src/index.rs [hunk 3]
```diff
@@ -34,12 +34,15 @@ use quickwit_common::GREEN_COLOR;
 use quickwit_config::{
```
Comment (fmassot) [inline quickwit-cli/src/index.rs:34]: Essentially modifications due to the move of code from quickwit-core to quickwit-control-plane.

**Comment by kstaken** [comment]
Is there a ticket that covers the design and reason for introducing the control plane service type and the plans for what it will solve? It looks like it's mirroring the metastore APIs, is it intended that nodes will talk to this service instead of talking to the metastore directly? Is it intended to be stateless? Will it be expected that you run more than one?

I know there's talk about having the merge planner run in this plane, what else is envisioned to live here?

**Comment by fmassot** [comment]
@kstaken some short answers so that you know a bit what's going on there:

> Is there a ticket that covers the design and reason for introducing the control plane service type and the plans for what it will solve?

yes I will document that very soon (hopefully by next week)

> It looks like it's mirroring the metastore APIs, is it intended that nodes will talk to this service instead of talking to the metastore directly? Is it intended to be stateless? Will it be expected that you run more than one?

Yes, this is correct. For the moment the plan is to have indexers making calls to the control plane that reads/writes to the metastore.

> I know there's talk about having the merge planner run in this plane, what else is envisioned to live here? 

We will run the merge planner there indeed. We will also handle indexing scheduling to decide which node should index a given (source_id, set of partition_ids). The solution should obviously work well with indexing with Kafka consumers. 

There is an open issue #1208 but I need to add there the current plan.

**Comment by fmassot** [comment]
Closing this old PR which is replaced by https://github.com/quickwit-oss/quickwit/pull/1853

**Pull request closed without merging**